src/Confluent.Kafka/Producer.cs

```diff
@@ -89,14 +89,14 @@ private Task StartPollTask(CancellationToken ct)
                         while (true)
                         {
                             ct.ThrowIfCancellationRequested();
-                            int eventsServedCountLocal = ownedKafkaHandle.Poll((IntPtr)cancellationDelayMaxMs);
+                            int eventsServedCount_ = ownedKafkaHandle.Poll((IntPtr)cancellationDelayMaxMs);
 
                             // note: lock {} is equivalent to Monitor.Enter then Monitor.Exit 
-                            lock (pollSyncObj)
+                            if (eventsServedCount_ > 0)
                             {
-                                if (eventsServedCountLocal > 0)
+                                lock (pollSyncObj)
                                 {
-                                    this.eventsServedCount = eventsServedCountLocal;
+                                    this.eventsServedCount = eventsServedCount_;
```

Comment:
this should be +=, see discussion on user-side of Poll().
I.e., number of events since last explicit Poll() since the number of events since last background Poll() isn't useful for the application to know